### PR TITLE
feat(scripts): Ability to specify target repo when installing

### LINKED ISSUE #126: Specify from which repo to install for use in scripting
Add the possibility to specify from which repo to install before searching then with `pacstall -I  repo-alias/package`

### PR DESCRIPTION
**Pull request by D-Brox**
## Purpose

Specify from which repo to install for use in scripting

## Approach

Add the possibility to specify from which repo to install before searching then with `pacstall -I package@user/reponame` for github and gitlab repos or `pacstall -I package@repourl` for other git-like repos.

## Addendum

Fixes #126

### CODE REVIEW

**Review by Henryws — APPROVED** [review]

**Review by wizard-28 — CHANGES_REQUESTED** [review]
I think we should log the place from where the package was installed in `$LOGDIR/metadata/pkg`. Also print it using the `-Qi` command.

---

Also display 
```
[*] INFO: Downloading <colored>pkg</colored> from <colored>`url of repo`</colored>
```
when downloading from a non-official repo

---

img

Ask for the other repos too. Don't call it `repo pacstall`, call it `official repo`.

---

img

Don't make it a `ask` just leave it as a `info`

File: misc/scripts/search.sh [hunk 1]
```diff
@@ -31,6 +31,47 @@ if [[ -n "$UPGRADE" ]]; then
...
+				error_log 3 "search $PACKAGE@$REPONAME"
+				return 1	
+			fi
+			export PACKAGE
+			export REPO=$URL
+			return 0
+		fi
+	done < "$STGDIR/repo/pacstallrepo.txt"
+	
+	fancy_message warn "There is no repo with the name $RED$REPONAME$NC"
```
Comment (wizard-28) [inline misc/scripts/search.sh:70]: img

```suggestion
	fancy_message warn "$RED$REPONAME$NC either doesn't exist or is not a pacstall programs repo"
```

I guess that makes more sense?

**Review by wizard-28 — CHANGES_REQUESTED** [review]

File: misc/scripts/search.sh (hunk 1 above)
Comment (D-Brox) [reply misc/scripts/search.sh:70]: I think "$REPONAME is not on your repo list or does not exist" would be better

Comment (wizard-28) [reply misc/scripts/search.sh:70]:
```suggestion
	fancy_message warn "$RED$REPONAME$NC is not on your repo list or doesn't exist"
```

Yeah.

**Review by wizard-28 — APPROVED** [review]

**Pull request merged**